var/spack/repos/builtin/packages/py-numba/package.py

```diff
@@ -53,6 +55,12 @@ class PyNumba(PythonPackage):
     depends_on("py-llvmlite@0.31", when="@0.48", type=("build", "run"))
     depends_on("py-importlib-metadata", when="@0.56:^python@:3.8", type=("build", "run"))
 
+    depends_on("tbb", when="+tbb")
+    conflicts("~tbb", when="@:0.50")  # No way to disable TBB
     # Version 6.0.0 of llvm had a hidden symbol which breaks numba at runtime.
     # See https://reviews.llvm.org/D44140
     conflicts("^llvm@6.0.0")
+
+    def setup_build_environment(self, env):
+        if self.spec.satisfies("~tbb"):
+            env.set("NUMBA_DISABLE_TBB", "yes")
```

Comment:
If we want to be really careful, we could also add:
```python
else:
    env.unset("NUMBA_DISABLE_TBB")
```
to ensure that existing env vars don't influence the build.